lightning-background-prober/src/lib.rs

```diff
@@ -0,0 +1,820 @@
+        Self {
+            path,
+            max_htlc_list: Vec::new(),
+        }
+    }
+
+    fn sorted_htlc_in_path(&mut self, channel_info: &ChannelInfo) {
+        self.max_htlc_list.clear(); // Clear the list before adding new values
+
+        for i in &self.path.hops {
+            let nodeid = NodeId::from_pubkey(&i.pubkey);
+            let max = channel_info.get_directional_info(0).unwrap().htlc_maximum_msat;
+            self.max_htlc_list.push((nodeid, max));
+        }
+        
+        self.max_htlc_list.sort_by(|(_, a), (_, b)| a.cmp(b)); // ascending ordering of MAX_HTLC
+    }
+}
+
+
+//Value selector for probing (Uses binary search method)
+async fn value_selector (
+	path: Path,
+	nodeid: NodeId,
+	probe_value_map: &ProbeValueMap,
+	max_htlc_list: Vec<(NodeId,u64)>,
+	capacity_list: Vec<(u64,Option<u64>)>,
+) -> u64 {
+	let mut value_map = probe_value_map.value_map.lock().await;
+	let temp = value_map.get_mut(&nodeid);
+	match temp {
+		Some(node_rating) => { 
+			
+			let vec_len = node_rating.probes.len();
+		
+			if vec_len == 0 {
+				let val = min(max_htlc_list[0].1, capacity_list[0].1.unwrap());
+				
+				node_rating.probes.push(ProbePersister::new(nodeid));
+				if let Some(last_probe) = node_rating.probes.last_mut() {
+					last_probe.update_path(path);
+					last_probe.update_value(val);
+				}
+				return val
+			}
+
+			
+			else if vec_len == 1 {
+				let probe_persister = node_rating.probes.last().unwrap();
+
+				let value = probe_persister.value.unwrap();
+				let probe_status = probe_persister.probe_status.as_ref().unwrap();
+				match probe_status {
+					
+					ProbeStatus::Success => {
+						let val = 2*value;
+
+						node_rating.probes.push(ProbePersister::new(nodeid));
+						if let Some(last_probe) = node_rating.probes.last_mut() {
+							last_probe.update_path(path);
+							last_probe.update_value(val);
+						}
+						return val
+					}
+					ProbeStatus::Failure => {
+						let val = (0.5*((value) as f64)).floor() as u64;
+						
+						node_rating.probes.push(ProbePersister::new(nodeid));
+						if let Some(last_probe) = node_rating.probes.last_mut() {
+							last_probe.update_path(path);
+							last_probe.update_value(val);
+						}
+						return val
+					}
+				}
+			}
+			
+			else {
+				let probe_persister = node_rating.probes.last().unwrap();
+
+				let value = probe_persister.value.unwrap();
+				let probe_status = probe_persister.probe_status.as_ref().unwrap();
+				match probe_status {
+					
+					ProbeStatus::Success => {
+						let val = 2*value;
+
+						node_rating.probes.push(ProbePersister::new(nodeid));
+						if let Some(last_probe) = node_rating.probes.last_mut() {
+							last_probe.update_path(path);
+							last_probe.update_value(val);
+						}
+						return val
+					}
+					ProbeStatus::Failure => {
+						let val_mid = node_rating.probes.get(vec_len-2).unwrap().value.unwrap();
+						let val = (0.5*((value + val_mid) as f64)).floor() as u64;
```

Comment:
Other than these lines, it looks like the `else` block is entirely the same as the preceding `else if` block. Could you write this to avoid repeating yourself?